### PR TITLE
feat(session): integrate scoped flow products

### LINKED ISSUE #175: Extend shared flow facts to relations and bindings
## Problem

The shared solver does not yet own all relation and binding-product facts needed by semantic consumers. Leaving those domains on legacy paths causes duplicate interpretation and can produce inconsistent identities or incomplete downstream results.

## Expected outcome

`FlowSlice` carries authoritative relation and binding-product facts with stable identity and deterministic semantics across all affected consumers.

## Acceptance

- Relation facts are represented and produced by the shared solver.
- Binding-product facts retain stable source and declaration identity.
- Consumers obtain these facts through the existing semantic query contract rather than local recomputation.
- Incremental and fresh evaluation produce equivalent relation and binding results.
- No parallel relation or binding resolver remains in production use.

AI-Generated

### PR DESCRIPTION
**Pull request by pikax**
Stacked on #507 (`tama_dag/D3P`), with the source prerequisite in #472. Review in dependency order and preserve the stack's required atomic landing; do not merge this PR on its own.

Routes the existing control interpreter through the shared graph-bound product kernel. Runtime reads consume exact prepared occurrences. Authored declarations and captured annotations retain original source identity and defining binder context; only selected source facts are hydrated.

The adapter transports existing continuation state through shared transfers and domain joins, preserving current narrowing, capture, and completion policies. Runtime alias declaration reads use the kernel's constant-time installed-authority index. Finally-write membership uses canonical identities in a hash set, eliminating growing prefix scans while preserving replay and fact order. Source hydration keeps exact authored lookups separate from runtime fallback.

Catch entry initializes only selected bindings. Generic-call arguments and receivers consume origins emitted by the shared value lowerer, sharing one exact identity lookup between value and freshness and avoiding a second wrapper classifier. Whole single-name typeof assertions read the exact queried binding's current flow type, independently of their operand and without runtime/effect/freshness credit. Selected assignment definitions use their retained expression-site address, so lookup work stays fixed as unrelated writes grow. Conditional-definition metadata preserves existing cold admission and unconditional-overwrite behavior while bindings keep their correct lexical identity.

Product-domain evidence flows through the existing report and finalizer. Missing required evidence keeps results non-publishable, and existing unsupported successor cases retain their typed gaps. Pre-existing structural completion and narrowing defects are documented with D4–D6 rather than implemented in this repair. The separate, pre-existing nominal signature constraint/default projection defect is recorded under its original nominal-projection owner.

The execution and determinism clauses now match that architecture. The live determinism test reverses equivalent demand and actual predecessor order and compares canonical product snapshots, exact executed-domain/subject evidence, the real discharge report, served results and warm reuse. Observation hooks compile only in unit tests. The owning charter records the sizing investigation, separating product cutover, stable-binding integration and nominal-output compatibility within the existing atomic candidate; it authorizes no successor work.

Final correction validation on `6340aedb1`: `node scripts/gate.mjs --exhaustive` passes with 25,133 core tests passed, 1,286 skipped, and all 27 WASM boundary tests passed. The previously failing `fallthrough_attrs_accepted_only_where_they_reach_the_dom` case passes in this complete run. Both strengthened determinism tests execute and pass. The focused missing-product/budget controls and strict roadmap validation also pass; a fresh independent review finds no actionable issues. The historical interaction did not reproduce; this does not claim its original runtime cause was identified.

Earlier validation: seven focused and 549 affected tests pass, as do all seven dedicated build/contract checks. Actual native and release WASM builds and all 762 affected package tests pass, with five native tests skipped. Four independent source reviews approve the scoped implementation.

Historical exhaustive run (before this correction): 25,128 core tests passed, one failed, 1,286 skipped, and all 27 WASM boundary tests passed. The sole failure is the scriptless-child unknown-attribute row in `cases::fallthrough::fallthrough_attrs_accepted_only_where_they_reach_the_dom`. All five exact isolated reruns pass; the repository classifier reports INTERACTION. The fixture and its production owner path are unchanged from the original top and the earlier passing checkpoint. The interaction's runtime cause remains undetermined, and the original full gate is recorded as failed. The standard triage CLI could not parse the gate's verdict block, so isolation reused its existing filter, argv and classification helpers and the gate's environment/supervision helpers with the named failure supplied directly. No test or tool expectations were changed.

The repository skips its shipped-cfg runtime lane; the core gate does not cover that lane. Full-workspace JavaScript verification remains blocked by unchanged conformance-harness import and Windows oracle-provisioning defects.

Closes #175

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved generated TypeScript declarations for Vue props, emits, models, and `defineExpose`.
  - Local symbols are rendered safely without invalid owner references, while imported symbols retain correct types and dependencies.
  - Improved handling of flow-sensitive values, assignments, captures, narrowing, and control-flow merges for more consistent analysis.
  - Preserved accurate metadata for call arguments, receivers, and `typeof` queries.

- **Tests**
  - Expanded regression coverage for nested symbols, flow analysis, capture behavior, generated declarations, and budget limits.

### CODE REVIEW

File: crates/verter_napi/src/host_compile_request_tests.rs [hunk 1]
```diff
@@ -33,6 +33,8 @@ fn identity() -> FfiHostCompileIdentity {
         component_id: Some("c-1".to_string()),
         is_production: false,
         force_js: false,
+        ssr_module_id: None,
+        hmr_strategy: None,
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_napi/src/host_compile_request_tests.rs:37]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: crates/verter_protocol/src/types.rs [hunk 2]
```diff
@@ -465,6 +465,32 @@ pub struct FfiSvelteCompileOptions {
...
+/// [`verter_compiler::compile_request::RuntimeStyleProcessing::wire_name`]:
+/// kebab-case, so `AuthoredOnly` is `"authored-only"`, not the `camelCase`
+/// container default's `authoredOnly`.
+#[derive(Deserialize, Debug, Clone, Copy, PartialEq, Eq, ts_rs::TS)]
+#[serde(rename_all = "camelCase")]
+#[ts(rename = "HostRuntimeStyleProcessing")]
+pub enum FfiRuntimeStyleProcessing {
+    Complete,
+    #[serde(rename = "authored-only")]
+    AuthoredOnly,
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_protocol/src/types.rs:479]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: crates/verter_wasm/src/host_compile_request_route_tests.rs [hunk 3]
```diff
@@ -659,36 +747,45 @@ fn the_vue_route_is_equivalent_to_the_legacy_pair_for_the_same_demand() {
...
-    assert_eq!(ide["code"], legacy["code"]);
-    assert_eq!(ide["sourceMap"], legacy["sourceMap"]);
-    assert_eq!(ide["isJsx"], legacy["isJsx"]);
-
-    // Diagnostics are named in the equivalence contract beside the bytes and
-    // the maps, so the whole set — codes, messages and spans — is compared
-    // against the legacy compile's rather than assumed to match.
-    let legacy_main = legacy_virtual_file(&host, "/src/App.vue", "main", None);
-    assert_eq!(response["diagnostics"], legacy_main["diagnostics"]);
+    assert_published_node_set(&no_map_response, &vue_app_expected_nodes(), false, &golden);
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_wasm/src/host_compile_request_route_tests.rs:780]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: packages/unplugin/src/core/preprocessor.ts [hunk 4]
```diff
@@ -136,9 +141,56 @@ export async function preprocessStyle(
...
+      sass = await import("sass");
+    } catch (e: unknown) {
+      if (e !== null && typeof e === "object" && "code" in e && e.code === "ERR_MODULE_NOT_FOUND") {
+        console.warn(
+          `[verter] Style preprocessing for lang="${lang}" requires the "sass" package. ` +
+            `Install it to compile ${lower} blocks outside Vite.`,
+        );
+        return null;
+      }
+      throw e;
```
[the remark this answers is not in the document]
Comment (pikax) [reply packages/unplugin/src/core/preprocessor.ts:161]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/compiler-compiler-bridge/CCA1O4.md [hunk 5]
```diff
@@ -64,3 +64,15 @@ Move unplugin's native host calls to CCA1O2's typed framework-discriminated requ
...
+
+The typed host-request route could not preserve Vue/Svelte bundling, HMR,
+and the manifest without three fields the typed wire schema did not carry
+(`ssr_module_id`, `hmr_strategy`, `style_processing`, plus the two enums
+they type), so the implementation extended the wire schema even though this
+charter's exclusions name native/NAPI signature work. The conflict, the
+footprint, and the ratification options are recorded in
+`decisions/2026-09-05-cca1o4-native-wire-extension-scope-disposition.md`.
+That record is a disclosure, not an approval; until the operator ruling the
+"Native/NAPI signatures … are excluded" line above stands as chartered.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/compiler-compiler-bridge/CCA1O4.md:78]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/compiler-compiler-core/CMP0.md [hunk 6]
```diff
@@ -68,7 +68,7 @@ Compiler request, policy, compatibility, and identity contract. The current owne
...
 
-- **CMP0-AC1 — sole-owner outcome:** the named final owner must be sole and every displaced route named below must be deleted or structurally rejected. Prefer existing type, capability, dependency, compiler, or static enforcement. Add a negative or mutation test only for a plausible critical fail-closed/correctness boundary or a reproduced defect that existing evidence does not discriminate.
+- **CMP0-AC1 — ownership contract:** enumerate every in-scope outcome, consumer and displaced production route. Each outcome/consumer has exactly one implementation owner; each displaced production route has exactly one later production-capable deletion/rejection owner. Bind owners to existing DAG node IDs, valid successor paths and receiving acceptance IDs under contracts/successor-charter-quality.md. The contract-owned schema/validator must reject missing inventory members, unknown/pathless owners and conflicting assignments before this node completes; production deletion remains acceptance of the later owner.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/compiler-compiler-core/CMP0.md:71]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/compiler-vue-compiler/VCP0.md [hunk 7]
```diff
@@ -84,7 +84,7 @@ Preflight evidence selection: preserve all four acceptance outcomes below, then
...
 - Correctness budget: zero stale publication, silent fallback, wrong-complete result, map/provenance loss, or identity aliasing.
-- Performance budget: when preflight identifies touched authority or a hot path, equivalent-work counters may increase by 0 and wall/allocation/RSS regression allowance remains 0.0% unless an owning-authority amendment supplies exact replacement thresholds. Otherwise performance evidence is not applicable; do not create counters or a 100-request retention soak solely to satisfy this charter.
+- Performance acceptance: use the exact applicable metric rows and methodology from performance-gates.toml, the applicable MEM0 budget, or the owning ratified product catalog, under contracts/resource-and-finalization.md (L2). Exact work invariants, latency/allocation/RSS limits under their owning methodology, and bounded new-capability budgets are distinct. New capabilities and deliberate pressure policies declare bounded new work and replacement SLOs before measurement. Missing required coverage needs an owning-contract amendment before measurement; no implicit 0.0% threshold or post-hoc rebaseline applies.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/compiler-vue-compiler/VCP0.md:87]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/compiler-vue-compiler/VCP5.md [hunk 8]
```diff
@@ -85,7 +85,7 @@ Preflight evidence selection: preserve all four acceptance outcomes below, then
...
 - Correctness budget: zero stale publication, silent fallback, wrong-complete result, map/provenance loss, or identity aliasing.
-- Performance budget: when preflight identifies touched authority or a hot path, equivalent-work counters may increase by 0 and wall/allocation/RSS regression allowance remains 0.0% unless an owning-authority amendment supplies exact replacement thresholds. Otherwise performance evidence is not applicable; do not create counters or a 100-request retention soak solely to satisfy this charter.
+- Performance acceptance: use the exact applicable metric rows and methodology from performance-gates.toml, the applicable MEM0 budget, or the owning ratified product catalog, under contracts/resource-and-finalization.md (L2). Exact work invariants, latency/allocation/RSS limits under their owning methodology, and bounded new-capability budgets are distinct. New capabilities and deliberate pressure policies declare bounded new work and replacement SLOs before measurement. Missing required coverage needs an owning-contract amendment before measurement; no implicit 0.0% threshold or post-hoc rebaseline applies.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/compiler-vue-compiler/VCP5.md:88]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/expansion-architecture-proof/ALPP.md [hunk 9]
```diff
@@ -84,7 +84,7 @@ Preflight evidence selection: preserve all four acceptance outcomes below, then
...
 - Correctness budget: zero stale publication, silent fallback, wrong-complete result, map/provenance loss, or identity aliasing.
-- Performance budget: when preflight identifies touched authority or a hot path, equivalent-work counters may increase by 0 and wall/allocation/RSS regression allowance remains 0.0% unless an owning-authority amendment supplies exact replacement thresholds. Otherwise performance evidence is not applicable; do not create counters or a 100-request retention soak solely to satisfy this charter.
+- Performance acceptance: use the exact applicable metric rows and methodology from performance-gates.toml, the applicable MEM0 budget, or the owning ratified product catalog, under contracts/resource-and-finalization.md (L2). Exact work invariants, latency/allocation/RSS limits under their owning methodology, and bounded new-capability budgets are distinct. New capabilities and deliberate pressure policies declare bounded new work and replacement SLOs before measurement. Missing required coverage needs an owning-contract amendment before measurement; no implicit 0.0% threshold or post-hoc rebaseline applies.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/expansion-architecture-proof/ALPP.md:87]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/expansion-architecture-proof/SLDP.md [hunk 10]
```diff
@@ -73,8 +73,8 @@ Preflight evidence selection: preserve all four acceptance outcomes below, then
 
 ## Deletions and forbidden designs
 
-- Delete or structurally reject: **framework-specific exception hidden in kernel**.
-- Delete or structurally reject: **parallel proof slices that contaminate evidence**.
+- Inventory and assign the later deletion/rejection owner for: **framework-specific exception hidden in kernel**. This node changes no production route.
+- Inventory and assign the later deletion/rejection owner for: **parallel proof slices that contaminate evidence**. This node changes no production route.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/expansion-architecture-proof/SLDP.md:78]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/expansion-cli/CLI0.md [hunk 11]
```diff
@@ -84,7 +84,7 @@ Preflight evidence selection: preserve all four acceptance outcomes below, then
...
 - Correctness budget: zero stale publication, silent fallback, wrong-complete result, map/provenance loss, or identity aliasing.
-- Performance budget: when preflight identifies touched authority or a hot path, equivalent-work counters may increase by 0 and wall/allocation/RSS regression allowance remains 0.0% unless an owning-authority amendment supplies exact replacement thresholds. Otherwise performance evidence is not applicable; do not create counters or a 100-request retention soak solely to satisfy this charter.
+- Performance acceptance: use the exact applicable metric rows and methodology from performance-gates.toml, the applicable MEM0 budget, or the owning ratified product catalog, under contracts/resource-and-finalization.md (L2). Exact work invariants, latency/allocation/RSS limits under their owning methodology, and bounded new-capability budgets are distinct. New capabilities and deliberate pressure policies declare bounded new work and replacement SLOs before measurement. Missing required coverage needs an owning-contract amendment before measurement; no implicit 0.0% threshold or post-hoc rebaseline applies.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/expansion-cli/CLI0.md:87]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/expansion-cli/CLITS0.md [hunk 12]
```diff
@@ -59,7 +59,7 @@ TypeScript-compatible `tsc` command. The current owner is **separate package lau
 - **Revision:** 4 — supersedes the 251-charter all-verticals proposal
 - **Prepared:** 2026-08-26
 - **Repository basis:** program/architecture-lock at d1f3d50a948597f036868543b9bb21acacd730ff
-- **Current-program condition:** maintainer work freeze; TCM0 = RESCOPE_REQUIRED; TCM1–TCM4 = LOCKED
+- **Current-program condition:** readiness follows the current DAG and implemented ledger; historical source-plan freeze/lock labels are not active dispatch instructions.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/expansion-cli/CLITS0.md:62]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/charters/expansion-kernel/CEF0.md [hunk 13]
```diff
@@ -60,13 +60,13 @@ Custom Element producer/consumer interoperability contract. The current owner is
...
 
-- **CEF0-AC1 — sole-owner outcome:** the named final owner must be sole and every displaced route named below must be deleted or structurally rejected. Prefer existing type, capability, dependency, compiler, or static enforcement. Add a negative or mutation test only for a plausible critical fail-closed/correctness boundary or a reproduced defect that existing evidence does not discriminate.
+- **CEF0-AC1 — ownership contract:** enumerate every in-scope outcome, consumer and displaced production route. Each outcome/consumer has exactly one implementation owner; each displaced production route has exactly one later production-capable deletion/rejection owner. Bind owners to existing DAG node IDs, valid successor paths and receiving acceptance IDs under contracts/successor-charter-quality.md. The contract-owned schema/validator must reject missing inventory members, unknown/pathless owners and conflicting assignments before this node completes; production deletion remains acceptance of the later owner.
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/charters/expansion-kernel/CEF0.md:69]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/decisions/2026-09-05-cca1o4-native-wire-extension-scope-disposition.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+
+- Status: proposed — awaiting operator/architect ratification
+- Date: 2026-09-05
+- Disposition: `ADOPT-NOW` for the native wire-schema extension and the
+  crates beyond CCA1O4's named surfaces, recorded here because the review
+  rounds found the expansion undispositioned, not because a fix owner may
+  ratify their own rescope.
+- Amends: nothing. No other charter's budget, no DAG edge, no other node's
+  ledger row. CCA1O4's own charter gains a pointer section to this record.
+- Asks for: one operator decision (see the last section).
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/decisions/2026-09-05-cca1o4-native-wire-extension-scope-disposition.md:11]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.

File: roadmap/0.1.0-tama/decisions/2026-09-05-cca1o4-native-wire-extension-scope-disposition.md [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+   `closure/typescript-mapper/closure.md` — the typescript-mapper closure
+   instrument's recorded `P-targeted-domain` proof and its
+   `CTL-targeted-selector` control. Five counts move (selected 9504 → 9510,
+   executed and passed 8957 → 8963, skipped unchanged at 547) and the
+   transcribed terminal summary is re-transcribed. Cause: this candidate and
+   the base each added `#[test]` cases inside the three packages that proof's
+   command selects, so the previously recorded counts no longer describe the
+   tree. This is the instrument's freshness obligation, not an evidence
+   relaxation — the comparison grammar, the skip basis, the declared 547
+   expected skips and the control's refusal semantics are unchanged, and
```
[the remark this answers is not in the document]
Comment (pikax) [reply roadmap/0.1.0-tama/decisions/2026-09-05-cca1o4-native-wire-extension-scope-disposition.md:126]: This thread is on content that entered this PR's diff only through the merge of the default branch: the file is byte-identical to `main`, and this branch does not change it. The PR's base has now been brought up to date, so the diff no longer carries it. Not this PR's change — resolving. If the finding is still live it belongs against `main`.
